### PR TITLE
add image feature extractors (RGB, ORB) #843

### PR DESCRIPTION
**Pull request by imaimai1125**
I have added an image feature plugin (ORB and RGB).
This plugin requires opencv 2.x or opencv 3.x.
(Related to #843.)

### CODE REVIEW

File: plugin/src/fv_converter/image_feature.cpp [hunk 1]
```diff
@@ -0,0 +1,162 @@
+// Jubatus: Online machine learning framework for distributed environment
+// Copyright (C) 2011 Preferred Networks and Nippon Telegraph and Telephone Corporation.
```
Comment (kmaehashi) [inline plugin/src/fv_converter/image_feature.cpp]: Copyright year needs to be updated for newly-added files.

File: plugin/src/fv_converter/image_feature.cpp [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+  x_size_(x_size),
+  y_size_(y_size) {
+  if (x_size_ <= 0 || y_size_<= 0) {
+    throw JUBATUS_EXCEPTION(
+        converter_exception("image size must be a positive number"));
+  }
+}
+
+void image_feature::dense_sampler(
+  const cv::Mat mat,
```
Comment (kmaehashi) [inline plugin/src/fv_converter/image_feature.cpp]: It is better to include OpenCV header file explicitly.

File: plugin/src/fv_converter/image_feature.cpp [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+}
+
+
+void image_feature::add_feature(
+  const std::string& key,
+  const std::string& value,
+  std::vector<std::pair<std::string, float> >& ret_fv) const {
+  std::vector<unsigned char> buf(value.begin(), value.end());
+
+#if(CV_MAJOR_VERSION == 3)
```
Comment (kmaehashi) [inline plugin/src/fv_converter/image_feature.cpp]: If ``CV_MAJOR_VERSION`` is not defined, the plugin should be compiled.  For example:

```
#ifndef CV_MAJOR_VERSION
#error "CV_MAJOR_VERSION is undefined"
#endif
```

**Comment by kmaehashi** [comment]
:+1: LGTM except for comments above.

File: plugin/src/fv_converter/image_feature.cpp (hunk 3 above)
Comment (kmaehashi) [inline plugin/src/fv_converter/image_feature.cpp]: In the offline discussion, ``CV_MAJOR_VERSION`` is a value defined in opencv header, so we don't need to  check this

File: plugin/src/fv_converter/image_feature.hpp [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+
+#ifndef JUBATUS_PLUGIN_FV_CONVERTER_IMAGE_FEATURE_HPP_
+#define JUBATUS_PLUGIN_FV_CONVERTER_IMAGE_FEATURE_HPP_
+
+#define CV_MAJOR_VERSION CV_MAJOR_VERSION
```
Comment (kmaehashi) [inline plugin/src/fv_converter/image_feature.hpp]: This line is unneeded.

File: plugin/src/fv_converter/image_feature.cpp [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+
+
+#include "image_feature.hpp"
+
```
Comment (kmaehashi) [inline plugin/src/fv_converter/image_feature.cpp]: It is better to ensure that CV_MAJOR_VERSION is 2 or 3 here?

**Comment by imaimai1125** [comment]
I reflected the comments.
In addition, I modified wscript to check the version of opencv at `./waf configure`

**Comment by kmaehashi** [comment]
:+1: Thank you!

Please add docs for this plugin in website repos.

**Pull request merged**